addons/stock/models/stock_quant.py

```diff
@@ -326,13 +369,50 @@ def action_inventory_history(self):
         return action
 
     def action_set_inventory_quantity(self):
+        quants_already_set = self.filtered(lambda quant: quant.inventory_quantity_set)
+        if quants_already_set:
+            ctx = dict(self.env.context or {})
+            ctx['default_quant_ids'] = self.ids
+            view = self.env.ref('stock.inventory_warning_set_view', False)
+            return {
+                'name': _('Quantities Already Set'),
+                'type': 'ir.actions.act_window',
+                'view_mode': 'form',
+                'views': [(view.id, 'form')],
+                'view_id': view.id,
+                'res_model': 'stock.inventory.warning',
+                'target': 'new',
+                'context': ctx,
+            }
         for quant in self:
+            quant.inventory_quantity_set = True
             quant.inventory_quantity = quant.quantity
         self.user_id = self.env.user.id
 
+    def action_reset(self):
+        ctx = dict(self.env.context or {})
+        ctx['default_quant_ids'] = self.ids
+        view = self.env.ref('stock.inventory_warning_reset_view', False)
+        return {
+            'name': _('Quantities To Reset'),
+            'type': 'ir.actions.act_window',
+            'view_mode': 'form',
+            'views': [(view.id, 'form')],
+            'view_id': view.id,
+            'res_model': 'stock.inventory.warning',
+            'target': 'new',
+            'context': ctx,
+        }
+
     def action_set_inventory_quantity_to_zero(self):
         self.inventory_quantity = 0
         self.inventory_diff_quantity = 0
+        self.inventory_quantity_set = False
+
+    def action_inventory_quantity_set(self):
+        for quant in self:
```

Comment:
As it's the same value for all the record in self, you can use a write() on self